src/v/cluster/partition_leaders_table.h

```diff
@@ -69,9 +93,17 @@ class partition_leaders_table {
       model::term_id term) {
         { f(tp_ns, pid, leader, term) } -> std::same_as<void>;
     }
-    void for_each_leader(Func&& f) const {
-        for (auto& [k, v] : _leaders) {
-            f(k.tp_ns, k.pid, v.current_leader, v.update_term);
+    ss::future<> for_each_leader(Func&& f) const {
+        auto version_snapshot = _version;
+        for (auto& [tp_ns, partition_leaders] : _topic_leaders) {
+            for (auto& [p_id, leader_info] : partition_leaders) {
+                f(tp_ns,
+                  p_id,
+                  leader_info.current_leader,
+                  leader_info.update_term);
+                co_await ss::coroutine::maybe_yield();
```

Comment:
Same `maybe_yield` comment applies here but not sure how often this function gets called.